### PR TITLE
Use peak_threads_usage instead of arrayUniq(thread_ids) in tests

### LINKED ISSUE #66116: Flaky `01275_parallel_mv`
Report: https://s3.amazonaws.com/clickhouse-test-reports/65838/13f4d0d7cfda499e44cbcabcdfda0557270fde04/stateless_tests__aarch64_.html

```
2024-07-04 17:55:15 --- /usr/share/clickhouse-test/queries/0_stateless/01275_parallel_mv.reference	2024-07-04 16:57:58.069466103 -0300
2024-07-04 17:55:15 +++ /tmp/clickhouse-test/0_stateless/01275_parallel_mv.gen.stdout	2024-07-04 17:55:15.777387568 -0300
2024-07-04 17:55:15 @@ -137,7 +137,7 @@
2024-07-04 17:55:15      Settings['parallel_view_processing'] = '1' and
2024-07-04 17:55:15      Settings['optimize_trivial_insert_select'] = '0' and
2024-07-04 17:55:15      Settings['max_insert_threads'] = '16';
2024-07-04 17:55:15 -18
2024-07-04 17:55:15 +10
2024-07-04 17:55:15  select count() from testX;
2024-07-04 17:55:15  60
2024-07-04 17:55:15  select count() from testXA;
2024-07-04 17:55:15 
2024-07-04 17:55:15 
2024-07-04 17:55:15 Settings used in the test: --max_insert_threads 16 --group_by_two_level_threshold 1 --group_by_two_level_threshold_bytes 48267495 --distributed_aggregation_memory_efficient 0 --fsync_metadata 0 --output_format_parallel_formatting 1 --input_format_parallel_parsing 0 --min_chunk_bytes_for_parallel_parsing 5854325 --max_read_buffer_size 927084 --prefer_localhost_replica 1 --max_block_size 27066 --max_joined_block_size_rows 72193 --max_threads 9 --optimize_append_index 0 --optimize_if_chain_to_multiif 0 --optimize_if_transform_strings_to_enum 0 --optimize_read_in_order 1 --optimize_or_like_chain 1 --optimize_substitute_columns 1 --enable_multiple_prewhere_read_steps 0 --read_in_order_two_level_merge_threshold 65 --optimize_aggregation_in_order 0 --aggregation_in_order_max_block_bytes 34059687 --use_uncompressed_cache 0 --min_bytes_to_use_direct_io 10737418240 --min_bytes_to_use_mmap_io 3012744388 --local_filesystem_read_method pread_threadpool --remote_filesystem_read_method read --local_filesystem_read_prefetch 1 --filesystem_cache_segments_batch_size 3 --read_from_filesystem_cache_if_exists_otherwise_bypass_cache 0 --throw_on_error_from_cache_on_write_operations 0 --remote_filesystem_read_prefetch 1 --allow_prefetched_read_pool_for_remote_filesystem 1 --filesystem_prefetch_max_memory_usage 32Mi --filesystem_prefetches_limit 0 --filesystem_prefetch_min_bytes_for_single_read_task 8Mi --filesystem_prefetch_step_marks 50 --filesystem_prefetch_step_bytes 100Mi --compile_aggregate_expressions 1 --compile_sort_description 0 --merge_tree_coarse_index_granularity 7 --optimize_distinct_in_order 1 --max_bytes_before_external_sort 10737418240 --max_bytes_before_external_group_by 10737418240 --max_bytes_before_remerge_sort 2240349947 --min_compress_block_size 1818230 --max_compress_block_size 1513298 --merge_tree_compact_parts_min_granules_to_multibuffer_read 113 --optimize_sorting_by_input_stream_properties 1 --http_response_buffer_size 1617984 --http_wait_end_of_query False --enable_memory_bound_merging_of_aggregation_results 1 --min_count_to_compile_expression 3 --min_count_to_compile_aggregate_expression 3 --min_count_to_compile_sort_description 0 --session_timezone America/Mazatlan --prefer_warmed_unmerged_parts_seconds 3 --use_page_cache_for_disks_without_file_cache True --page_cache_inject_eviction True --merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability 0.81 --prefer_external_sort_block_bytes 1 --cross_join_min_rows_to_compress 1 --cross_join_min_bytes_to_compress 100000000 --min_external_table_block_size_bytes 1 --max_parsing_threads 1
2024-07-04 17:55:15 
2024-07-04 17:55:15 MergeTree settings used in test: --ratio_of_defaults_for_sparse_serialization 0.0 --prefer_fetch_merged_part_size_threshold 8755514578 --vertical_merge_algorithm_min_rows_to_activate 1000000 --vertical_merge_algorithm_min_columns_to_activate 1 --allow_vertical_merges_from_compact_to_wide_parts 1 --min_merge_bytes_to_use_direct_io 668520075 --index_granularity_bytes 12693347 --merge_max_block_size 13065 --index_granularity 3514 --min_bytes_for_wide_part 232735480 --marks_compress_block_size 25279 --primary_key_compress_block_size 67609 --replace_long_file_name_to_hash 0 --max_file_name_length 0 --min_bytes_for_full_part_storage 45635382 --compact_parts_max_bytes_to_buffer 83520586 --compac
```

### PR DESCRIPTION
**Pull request by azat**
Should fix flakiness like [1].

  [1]: https://s3.amazonaws.com/clickhouse-test-reports/66098/93afc8e6133365007488c4d8340f434f6e8a876f/stateless_tests__aarch64_.html

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fixes: https://github.com/ClickHouse/ClickHouse/issues/66116

#### CI Settings (Only check the boxes if you know what you are doing):
- [x]  Allow: Stateless tests

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit daae5d4d4661c780b6368950ec484415ca3a0492 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66162/daae5d4d4661c780b6368950ec484415ca3a0492/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/integration_tests__aarch64__[1_6].html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/76119a4567ce2ac9c0aff715c1a9ba2607e806e0/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by alexey-milovidov** [comment]
@azat, Could you please merge with the master branch? We upgraded the compiler and it started to find one more issue by clang-tidy, which was fixed in the master branch.

**Comment by azat** [comment]
@alexey-milovidov the compiler hadn't been upgraded yet, it is done in https://github.com/ClickHouse/ClickHouse/pull/66070 and it hasn't been merged yet, it also fixes the tidy issues

**Comment by azat** [comment]
>Stateless tests (aarch64) — fail: 1, passed: 6701, skipped: 17

- `01563_distributed_query_finish` - https://github.com/ClickHouse/ClickHouse/pull/66171

P.S. And this PR should be created with `Allow: Stateless tests` only I guess to avoid this annoying failed checks

File: tests/queries/0_stateless/01275_parallel_mv.reference [hunk 1]
```diff
@@ -10,14 +10,14 @@ insert into testX select number from numbers(10) settings
...
-select arrayUniq(thread_ids) from system.query_log where
+select peak_threads_usage from system.query_log where
     current_database = currentDatabase() and
     type != 'QueryStart' and
     query like '%insert into testX %' and
     Settings['parallel_view_processing'] = '0' and
     Settings['optimize_trivial_insert_select'] = '0' and
     Settings['max_insert_threads'] = '0';
-2
+5
```
Comment (CheSema) [inline tests/queries/0_stateless/01275_parallel_mv.reference:20]: I do not understand why. Why count all participated threads bigger that maximum simultaneous running threads?
Do we sure that it is counted correctly?

Comment (CheSema) [reply tests/queries/0_stateless/01275_parallel_mv.reference:20]: https://github.com/ClickHouse/ClickHouse/issues/66116#issuecomment-2211753262

Comment (CheSema) [reply tests/queries/0_stateless/01275_parallel_mv.reference:20]: I would rather wait with merging this fix until we understand it.
May be we need fix the counter instead using wrong one as a fix for something.

**Comment by alexey-milovidov** [comment]
We want to get rid of all flaky tests. It makes sense to run even trivial PRs with all checks to find out if some tests are still flaky and then have a huge pain fixing all of them.

File: tests/queries/0_stateless/01275_parallel_mv.reference (hunk 1 above)
Comment (CheSema) [reply tests/queries/0_stateless/01275_parallel_mv.reference:20]: Fix https://github.com/ClickHouse/ClickHouse/pull/66230

Comment (azat) [reply tests/queries/0_stateless/01275_parallel_mv.reference]: Rebased on top of #66230

**Review by CheSema — APPROVED** [review]

**Pull request merged**